packages/next/src/client/resolve-href.ts

```diff
@@ -34,7 +34,8 @@ export function resolveHref(
 
   // repeated slashes and backslashes in the URL are considered
   // invalid and will never match a Next.js page/file
-  const urlProtoMatch = urlAsString.match(/^[a-zA-Z]{1,}:\/\//)
+  // https://datatracker.ietf.org/doc/html/rfc1738#section-2.1
+  const urlProtoMatch = urlAsString.match(/^[a-zA-Z][a-zA-Z+.-]*:\/\//)
```

Comment:
If I read https://datatracker.ietf.org/doc/html/rfc1738#section-5 right, this should also allow digits

Reply:
https://www.rfc-editor.org/rfc/rfc3986.html#section-3.1 specifies that it has to start with a letter. `new URL('1://google.com')` would throw so switching to https://www.rfc-editor.org/rfc/rfc3986.html#section-3.1

Reply:
3986 also makes sense because it supersedes 1738.

Reply:
Looks good now. Slightly worried about codebases where someone might have used uppercase for whatever reason.

Reply:
Sorry, I missed the `i` in the regex to make it case insensitive (and your test case)

Reply:
Yeah, felt like it matches closer what the spec say "alphanumeric but also you should be case-insensitive"